spec/features/users_pages_spec.rb

```diff
@@ -65,4 +66,35 @@
       expect(page).to have_title("Sign up | Ruby on Rails Tutorial Sample App")
     end
   end
+
+
+
+  describe "All users" do
+    before do
+      30.times { FactoryGirl.create(:user) }
+      visit login_path
+      fill_in 'Email', with: admin.email
+      fill_in 'Password', with: admin.password
+      click_button 'Log in'
+      visit users_path
+    end
+
+    it 'including pagination' do
+      expect(page).to have_selector('div.pagination')
+      expect(page).to have_selector('a', text: 'Next')
+
+      User.paginate(page: 1).each do |user|
+        expect(page).to have_selector('a', text: user.name)
+      end
+    end
+
+    it 'delete users' do
+      expect do
+        click_link('delete', match: :first)
```

Comment:
`match: :first` might give you unexpected results. It's always better to exactly know what you click on.